app/src/main/java/com/islaharper/dawnofandroid/presentation/screens/splash/SplashScreen.kt

```diff
@@ -1,15 +1,80 @@
 package com.islaharper.dawnofandroid.presentation.screens.splash
 
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.height
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
 import androidx.compose.runtime.collectAsState
 import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
 import androidx.hilt.navigation.compose.hiltViewModel
 import androidx.navigation.NavHostController
+import com.airbnb.lottie.LottieComposition
+import com.airbnb.lottie.compose.LottieAnimation
+import com.airbnb.lottie.compose.LottieCompositionSpec
+import com.airbnb.lottie.compose.animateLottieCompositionAsState
+import com.airbnb.lottie.compose.rememberLottieComposition
+import com.islaharper.dawnofandroid.R
+import com.islaharper.dawnofandroid.navigation.Screen
+import com.islaharper.dawnofandroid.ui.theme.SPLASH_SPACER_HEIGHT
+import com.islaharper.dawnofandroid.ui.theme.infoGreen
 
 @Composable
 fun SplashScreen(
     navHostController: NavHostController,
     splashViewModel: SplashViewModel = hiltViewModel(),
 ) {
     val onBoardingComplete by splashViewModel.onBoardingCompleted.collectAsState()
+    val composition by rememberLottieComposition(spec = LottieCompositionSpec.RawRes(R.raw.jetpack))
+    val progress by animateLottieCompositionAsState(composition = composition)
+
+    LaunchedEffect(key1 = progress) {
+        if (progress == 1f) {
+            navHostController.popBackStack()
+            if (onBoardingComplete) {
+                navHostController.navigate(Screen.Login.route)
+            } else {
+                navHostController.navigate(Screen.Welcome.route)
+            }
+        }
+    }
+    Splash(composition, progress)
+}
+
+@Composable
+private fun Splash(composition: LottieComposition?, progress: Float) {
+    Box(modifier = Modifier.fillMaxSize()) {
+        LottieAnimation(composition = composition, progress = progress)
+        Column(
+            modifier = Modifier.fillMaxSize(),
+            verticalArrangement = Arrangement.Center,
+            horizontalAlignment = Alignment.CenterHorizontally,
+        ) {
+            Spacer(modifier = Modifier.height(SPLASH_SPACER_HEIGHT))
+            Text(
+                text = stringResource(R.string.app_name),
+                fontSize = MaterialTheme.typography.h4.fontSize,
+                fontWeight = FontWeight.Bold,
+                color = MaterialTheme.colors.infoGreen,
```

Comment:
Why have we created our own extension property `infoGreen` on MaterialTheme? Is this following a pattern in an article or a course? Could it not be a theme color (like primary etc..)? 

Similar for the fonts, have a look at the way NowInAndroid does it:
https://github.com/android/nowinandroid/blob/main/core/designsystem/src/main/java/com/google/samples/apps/nowinandroid/core/designsystem/theme/Theme.kt

https://github.com/android/nowinandroid/blob/main/core/designsystem/src/main/java/com/google/samples/apps/nowinandroid/core/designsystem/theme/Type.kt

Reply:
I added the custom colours as I like being concise, I suppose I could call the colours directly for specific cases that don't have light / dark variants eg: color = InfoGreen, and see if the colours that have variants I can reuse MaterialTheme values such as secondary and secondaryVariant etc

Reply:
I see, so this color doesn't vary for light dark? Will it be used in more than one place? If not, you can probably just inline it. If it will be reused, I don't think it belongs as an extension property on the theme colors - that is confusing because theme colors are supposed to be named semantically (onPrimary, primary etc.). 

I think it's worth spending some more time here. Getting the theme right makes the difference between an Android app that is mediocre and an Android app that is great through supporting light/dark and Material You. Clients have asked us for this in the past. 

Same goes for the typography, it can look sloppy when each screen has its own non-standard font.

Reply:
Refactored theme and colours, added data store for dark mode and dynamic theme for use later in project.